forem/forem
# Implement Unified Embeds for StackExchange and StackOverflow URLs

## What type of PR is this? (check all applicable)


## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Stackexchange and Stackoverflow embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15578

## QA Instructions, Screenshots, Recordings
Example Stackexchange and Stackoverflow URLs:
https://travel.stackexchange.com/questions/172014/is-it-okay-to-mix-in-local-language-when-i-know-it-poorly
https://diy.stackexchange.com/q/244088
https://academia.stackexchange.com/a/181893
https://stackoverflow.com/q/70974409/9091371
https://stackoverflow.com/questions/70976451/changing-h1-element-based-on-input-element-content-vanilla-js
https://stackoverflow.com/a/70976251/9091371

Example Stackexchange and Stackoverflow IDs:
`{% stackexchange 172014 travel %}`
`{% stackexchange 244088 diy %}`
`{ % stackoverflow 70974409 %}`
`{% stackoverflow 70976251 %}`

- As a user, create the appropriate Stackoveflow or Stackexchange liquid tag using this format: `{% embed <relevant-url> %}`. The Liquid Tag should render properly.
- Also create liquid tags using the old method: `{% <relevant-keyword> <relevant-id> <optional-site> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?
None

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.

## [optional] Are there any post deployment tasks we need to perform?
None
